upgradability/set_code_hash/voting_fixed/README.md

```diff
@@ -0,0 +1,70 @@
+# Voting fixed example
+
+This smart contract is part of an example showing how we can write upgradable smart contracts in ink! using `set_code_hash`.
+
+See `voting` and `transparent_voting` examples first.
+
+## Description
+
+This contract provides a simple voting interface which allows voting for one of some two options. However, when compared to
+`voting`, this contract has ability to store more votes and forbids adding new voters or
+upgrading itself.
+
+The goal of this example is to show how one might perform a smart contract upgrade (using `set_code_hash` method) in a case in which
+storage layouts of the original contract and the new one differ.
+
+## Upgrading `voting` smart contract: upgrade with storage layout change
+
+For an introduction of how to upgrade with `set_code_hash` see `transparent_voting` example.
+
+### Example scenario
+
+Let's say you deployed `voting` contract, updated list of the allowed voters to it's
+desired state and gathered some votes. Then you realized that the contract has a flaw:
+votes are stored in `u8` counters, and will soon overflow.
+
+Now, you want to fix this issue changing the counters to `u64`, and remove ability
+to add new voters/upgrade the contract (to assure users that the list of voters is final).

```

Comment:
Maybe instead of removing the possibility to add new voters we could add a check that `voters.len() < u64::MAX`?

Reply:
Hmm, exceeding `u64::MAX` does not seem to be realistic, the idea here was to showcase that we can upgrade contract to a version in which some property of this contract becomes immutable - here we can no longer change the set of the voters, so the users of the contract see that we cannot influence the vote (even as the creator of the contract).

Reply:
Removing possibility to add new voters seems a little bit weird that's why I proposed to check if there is a space for a new voter. However, I agree that exceeding u64::MAX does not seem to be realistic. Then, either leave the example as it is or consider adding a parameter `voters_limit`.

Reply:
To be honest, I would prefer to leave it as it is.